mm2src/coins/nft/nft_errors.rs

```diff
@@ -21,15 +23,38 @@ pub enum GetNftInfoError {
     Internal(String),
     GetEthAddressError(GetEthAddressError),
     #[display(
-        fmt = "Token: token_address {}, token_id {} was not find in wallet",
+        fmt = "Token: token_address {}, token_id {} was not found in wallet",
         token_address,
         token_id
     )]
     TokenNotFoundInWallet {
         token_address: String,
         token_id: String,
     },
-    AddressError(String),
+    #[display(fmt = "DB error {}", _0)]
+    DbError(String),
+    #[display(
+        fmt = "Error parsing datetime to timestamp. Expected format 'YYYY-MM-DDTHH:MM:SS.sssZ', got: {}",
+        _0
+    )]
+    ParseTimestampError(String),
+    #[display(fmt = "The contract type is required and should not be null.")]
+    ContractTypeIsNull,
+}
+
+impl From<GetNftInfoError> for WithdrawError {
+    fn from(e: GetNftInfoError) -> Self { WithdrawError::GetNftInfoError(e) }
+}
+
+impl From<SlurpError> for GetNftInfoError {
+    fn from(e: SlurpError) -> Self {
+        let error_str = e.to_string();
+        match e {
+            SlurpError::ErrorDeserializing { .. } => GetNftInfoError::InvalidResponse(error_str),
+            SlurpError::Transport { .. } | SlurpError::Timeout { .. } => GetNftInfoError::Transport(error_str),
+            SlurpError::Internal(_) | SlurpError::InvalidRequest(_) => GetNftInfoError::Internal(error_str),
```

Comment:
:thinking: well, `SlurpError` doesnt have `InvalidResponse` but yeah, we match it with `InvalidResponse` or `Transport`.
As for `SlurpError::InvalidRequest` sometimes we match it with `InvalidRequest` or with `Internal` . I think bcz we use custom `slurp_req` functions, and `slurp_req_body` can propagate `SlurpError::Internal` error.
But  yeah, I think its better to use `InvalidRequest` matching to avoid misunderstanding.

Changed matches for `GetNftInfoError` and `GetInfoFromUriError`

Reply:
great! thanks